src/diffusers/models/unets/unet_motion_model.py

```diff
@@ -260,13 +280,26 @@ def __init__(
         if encoder_hid_dim_type is None:
             self.encoder_hid_proj = None
 
+        if addition_embed_type == "text_time":
+            self.add_time_proj = Timesteps(addition_time_embed_dim, True, 0)
+            self.add_embedding = TimestepEmbedding(projection_class_embeddings_input_dim, time_embed_dim)
```

Comment:
Could you also load add_embedding in from_unet2d? Something like:

```
if hasattr(model, "add_embedding"):
    model.add_embedding.load_state_dict(unet.add_embedding.state_dict())
```

Reply:
If we take a look at current unet_2d_condition.py modelling code, the team has refactored out these changes into separate helper functions. I think that because unet_motion_model.py is mostly a copy of that, we can adapt those changes here and therefore all the functionality one would need. We can take it up in a future PR in my opinion (also I'm afraid I will not have time to test things thoroughly).